lineage/BaumWelch.py

```diff
@@ -147,33 +146,31 @@ def do_M_E_step(tHMMobj, gammas):
     for state_j in range(tHMMobj.num_states):
         tHMMobj.estimate.E[state_j].estimator(all_cells, all_gammas[:, state_j])
 
-
 def do_M_E_step_1state(all_tHMMobj, gammas_1st, state_j):
     """ perform the M_E step when all the concentrations are given at once for one state.
     all_cells is a list (for different concentrations) containing lists of tuples of observations. 
     gammas_1st is a list of lists, each cell's gamma value for one state. """
-    all_cells = []
+    G1cells = []
+    G2cells = []
     for tHMMobj in all_tHMMobj:
-        all_cells.append([cell.obs for lineage in tHMMobj.X for cell in lineage.output_lineage])
-        # TODO: the line where we pass the cells and gammas to the atonce_estimator
-        for i in range(len(tHMMobj.estimate.E[state_j].params)):
-            tHMMobj.estimate.E[state_j].params[i] = 0.5
+        all_cells = np.array([cell.obs for lineage in tHMMobj.X for cell in lineage.output_lineage])
+        G1cells.append(all_cells[:, np.array([0, 2, 4])])
+        G2cells.append(all_cells[:, np.array([1, 3, 5])])
```

Comment:
Run `atonce_estimator` once outside here, then just assign the results to the various emissions by directly writing to `tHMMobj.estimate.E[state_j].params`. There's no need to write to params from within the class.

Reply:
Also, are you sure that `do_M_pi_step` isn't being called when you do the combined fitting? That would overwrite your results.

Reply:
Thank you. I will bring that back.
Why would `do_M_pi_step` overwrite the parameters? I didn't see anything being assigned to the tHMMobj in `do_M_pi_step` or `do_M_T_step`. Could you elaborate please?

Reply:
Oh, I meant `do_M_E_step`.

Reply:
I have a separate `do_M_E_step` that when fitting at once, it is called.
I will check again. Thank you.